terraform-provider-azurerm_v3.1.0_x5 plugin - panic: runtime error

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

<!--- Please keep this note for the community --->

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->


### Terraform Version

1.1.8

### AzureRM Provider Version

3.1.0

### Affected Resource(s)/Data Source(s)

azurerm_v3.1.0_x5

### Terraform Configuration Files

```hcl
resource "azurerm_resource_group" "resource_group" {
    name     = "${var.application}-${var.environment}-rg"
    location = var.location
    
    tags = var.tags
}


resource "azurerm_service_plan" "example" {
  name                = local.AppPlanName
  location            = var.location
  resource_group_name = azurerm_resource_group.resource_group.name
  os_type             = "Linux"
  sku_name            = var.webappskusize
  tags = var.tags
}

resource "azurerm_linux_web_app" "example_app" {
  name                    = local.example_appname
  location                = var.location
  resource_group_name     = azurerm_resource_group.resource_group.name
  service_plan_id         = azurerm_service_plan.example.id
  client_affinity_enabled = false
  

  tags = var.tags

  identity {
    type = "SystemAssigned"
  }

  site_config {
    always_on                 = true
    http2_enabled             = true
    use_32_bit_worker         = false
            
   }
}
```


### Debug Output/Panic Output

```shell
Stack trace from the terraform-provider-azurerm_v3.1.0_x5 plugin:

panic: runtime error: index out of range [0] with length 0

goroutine 533 [running]:
github.com/hashicorp/terraform-provider-azurerm/internal/services/appservice/helpers.ExpandSiteConfigLinux({0xc00017d6c0, 0x3?, 0x6fc23ac00?}, 0x0, {0xc001464380, {0x63a73f0, 0xc00035f458}, 0xc00222d280, 0x0, {0x63a8ff0, ...}})
/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/internal/services/appservice/helpers/web_app_schema.go:2931 +0x141d
github.com/hashicorp/terraform-provider-azurerm/internal/services/appservice.LinuxWebAppResource.Create.func1({0x63a5e10, 0xc002135440}, {0xc001464380, {0x63a73f0, 0xc00035f458}, 0xc00222d280, 0x0, {0x63a8ff0, 0xa24d7e8}})
/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/internal/services/appservice/linux_web_app_resource.go:277 +0xbc5
github.com/hashicorp/terraform-provider-azurerm/internal/sdk.(*ResourceWrapper).Resource.func2({0x63a5e10, 0xc002135440}, 0x1a3185c4557?, {0x5382d00?, 0xc001464380?})
/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/internal/sdk/wrapper_resource.go:52 +0x163
github.com/hashicorp/terraform-provider-azurerm/internal/sdk.diagnosticsWrapper.func1({0x63a5e10, 0xc002135440}, 0x53bb1e0?, {0x5382d00, 0xc001464380})
/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/internal/sdk/wrapper_resource.go:185 +0xbe
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).create(0xc0003c5ce0, {0x63a5da0, 0xc00221e480}, 0x2?, {0x5382d00, 0xc001464380})
/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema/resource.go:341 +0x12e
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc0003c5ce0, {0x63a5da0, 0xc00221e480}, 0xc002258000, 0xc00222cc00, {0x5382d00, 0xc001464380})
/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema/resource.go:467 +0x6f1
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc000699698, {0x63a5da0, 0xc00221e480}, 0xc002140e60)
/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema/grpc_provider.go:977 +0xd8a
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0xc00012ae00, {0x63a5e48?, 0xc0022202a0?}, 0xc00061b880)
/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server/server.go:603 +0x2ad
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0x58cea60?, 0xc00012ae00}, {0x63a5e48, 0xc0022202a0}, 0xc002134de0, 0x0)
/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:380 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc00017d180, {0x63b6db0, 0xc0000f6900}, 0xc0021f6900, 0xc000c3fc50, 0xa20c980, 0x0)
/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:1292 +0xc6f
google.golang.org/grpc.(*Server).handleStream(0xc00017d180, {0x63b6db0, 0xc0000f6900}, 0xc0021f6900, 0x0)
/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:1617 +0xa1b
google.golang.org/grpc.(*Server).serveStreams.func1.2()
/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:940 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-azurerm/vendor/google.golang.org/grpc/server.go:938 +0x28a

Error: The terraform-provider-azurerm_v3.1.0_x5 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```


### Expected Behaviour

Resources to be created with out error 

### Actual Behaviour

Panic attack : runtime error during terraform apply 

### Steps to Reproduce

terraform apply

### Important Factoids

_No response_

### References

_No response_

--- a/internal/services/appservice/helpers/web_app_schema.go
+++ b/internal/services/appservice/helpers/web_app_schema.go
@@ -2927,7 +2927,7 @@ func ExpandSiteConfigLinux(siteConfig []SiteConfigLinux, existing *web.SiteConfi
 		expanded.AppCommandLine = utils.String(linuxSiteConfig.AppCommandLine)
 	}
 
-	if metadata.ResourceData.HasChange("site_config.0.application_stack") {
+	if metadata.ResourceData.HasChange("site_config.0.application_stack") && len(linuxSiteConfig.ApplicationStack) > 0 {
 		linuxAppStack := linuxSiteConfig.ApplicationStack[0]
 		if linuxAppStack.NetFrameworkVersion != "" {
 			expanded.LinuxFxVersion = utils.String(fmt.Sprintf("DOTNETCORE|%s", linuxAppStack.NetFrameworkVersion))

--- a/internal/services/appservice/helpers/web_app_schema.go
+++ b/internal/services/appservice/helpers/web_app_schema.go
@@ -880,6 +880,14 @@ func windowsApplicationStackSchema() *pluginsdk.Schema {
 				"dotnet_version": {
 					Type:     pluginsdk.TypeString,
 					Optional: true,
+					AtLeastOneOf: []string{
+						"site_config.0.application_stack.0.dotnet_version",
+						"site_config.0.application_stack.0.php_version",
+						"site_config.0.application_stack.0.python_version",
+						"site_config.0.application_stack.0.node_version",
+						"site_config.0.application_stack.0.java_version",
+						"site_config.0.application_stack.0.docker_container_name",
+					},
 					ValidateFunc: validation.StringInSlice([]string{
 						"v3.0",
 						"v4.0",
@@ -891,6 +899,14 @@ func windowsApplicationStackSchema() *pluginsdk.Schema {
 				"php_version": {
 					Type:     pluginsdk.TypeString,
 					Optional: true,
+					AtLeastOneOf: []string{
+						"site_config.0.application_stack.0.dotnet_version",
+						"site_config.0.application_stack.0.php_version",
+						"site_config.0.application_stack.0.python_version",
+						"site_config.0.application_stack.0.node_version",
+						"site_config.0.application_stack.0.java_version",
+						"site_config.0.application_stack.0.docker_container_name",
+					},
 					ValidateFunc: validation.StringInSlice([]string{
 						"7.4",
 					}, false),
@@ -899,6 +915,14 @@ func windowsApplicationStackSchema() *pluginsdk.Schema {
 				"python_version": {
 					Type:     pluginsdk.TypeString,
 					Optional: true,
+					AtLeastOneOf: []string{
+						"site_config.0.application_stack.0.dotnet_version",
+						"site_config.0.application_stack.0.php_version",
+						"site_config.0.application_stack.0.python_version",
+						"site_config.0.application_stack.0.node_version",
+						"site_config.0.application_stack.0.java_version",
+						"site_config.0.application_stack.0.docker_container_name",
+					},
 					ValidateFunc: validation.StringInSlice([]string{
 						"3.4.0", // Note: The portal only lists `3.6`, however, the service only uses the value `3.4.0`. Anything else is silently discarded
 					}, false),
@@ -907,6 +931,14 @@ func windowsApplicationStackSchema() *pluginsdk.Schema {
 				"node_version": { // Discarded by service if JavaVersion is specified
 					Type:     pluginsdk.TypeString,
 					Optional: true,
+					AtLeastOneOf: []string{
+						"site_config.0.application_stack.0.dotnet_version",
+						"site_config.0.application_stack.0.php_version",
+						"site_config.0.application_stack.0.python_version",
+						"site_config.0.application_stack.0.node_version",
+						"site_config.0.application_stack.0.java_version",
+						"site_config.0.application_stack.0.docker_container_name",
+					},
 					ValidateFunc: validation.StringInSlice([]string{
 						"12-LTS",
 						"14-LTS",
@@ -920,6 +952,14 @@ func windowsApplicationStackSchema() *pluginsdk.Schema {
 				"java_version": {
 					Type:     pluginsdk.TypeString,
 					Optional: true,
+					AtLeastOneOf: []string{
+						"site_config.0.application_stack.0.dotnet_version",
+						"site_config.0.application_stack.0.php_version",
+						"site_config.0.application_stack.0.python_version",
+						"site_config.0.application_stack.0.node_version",
+						"site_config.0.application_stack.0.java_version",
+						"site_config.0.application_stack.0.docker_container_name",
+					},
 					ValidateFunc: validation.StringInSlice([]string{
 						"1.8",
 						"11",
@@ -948,8 +988,16 @@ func windowsApplicationStackSchema() *pluginsdk.Schema {
 				},
 
 				"docker_container_name": {
-					Type:         pluginsdk.TypeString,
-					Optional:     true,
+					Type:     pluginsdk.TypeString,
+					Optional: true,
+					AtLeastOneOf: []string{
+						"site_config.0.application_stack.0.dotnet_version",
+						"site_config.0.application_stack.0.php_version",
+						"site_config.0.application_stack.0.python_version",
+						"site_config.0.application_stack.0.node_version",
+						"site_config.0.application_stack.0.java_version",
+						"site_config.0.application_stack.0.docker_container_name",
+					},
 					ValidateFunc: validation.StringIsNotEmpty,
 					RequiredWith: []string{
 						"site_config.0.application_stack.0.docker_container_tag",
@@ -1078,6 +1126,15 @@ func linuxApplicationStackSchema() *pluginsdk.Schema {
 				"dotnet_version": {
 					Type:     pluginsdk.TypeString,
 					Optional: true,
+					AtLeastOneOf: []string{
+						"site_config.0.application_stack.0.dotnet_version",
+						"site_config.0.application_stack.0.php_version",
+						"site_config.0.application_stack.0.python_version",
+						"site_config.0.application_stack.0.node_version",
+						"site_config.0.application_stack.0.ruby_version",
+						"site_config.0.application_stack.0.java_version",
+						"site_config.0.application_stack.0.docker_image",
+					},
 					ValidateFunc: validation.StringInSlice([]string{
 						"3.1",
 						"5.0",
@@ -1095,6 +1152,15 @@ func linuxApplicationStackSchema() *pluginsdk.Schema {
 				"php_version": {
 					Type:     pluginsdk.TypeString,
 					Optional: true,
+					AtLeastOneOf: []string{
+						"site_config.0.application_stack.0.dotnet_version",
+						"site_config.0.application_stack.0.php_version",
+						"site_config.0.application_stack.0.python_version",
+						"site_config.0.application_stack.0.node_version",
+						"site_config.0.application_stack.0.ruby_version",
+						"site_config.0.application_stack.0.java_version",
+						"site_config.0.application_stack.0.docker_image",
+					},
 					ValidateFunc: validation.StringInSlice([]string{
 						"7.4",
 						"8.0",
@@ -1111,6 +1177,15 @@ func linuxApplicationStackSchema() *pluginsdk.Schema {
 				"python_version": {
 					Type:     pluginsdk.TypeString,
 					Optional: true,
+					AtLeastOneOf: []string{
+						"site_config.0.application_stack.0.dotnet_version",
+						"site_config.0.application_stack.0.php_version",
+						"site_config.0.application_stack.0.python_version",
+						"site_config.0.application_stack.0.node_version",
+						"site_config.0.application_stack.0.ruby_version",
+						"site_config.0.application_stack.0.java_version",
+						"site_config.0.application_stack.0.docker_image",
+					},
 					ValidateFunc: validation.StringInSlice([]string{
 						"3.7",
 						"3.8",
@@ -1128,6 +1203,15 @@ func linuxApplicationStackSchema() *pluginsdk.Schema {
 				"node_version": {
 					Type:     pluginsdk.TypeString,
 					Optional: true,
+					AtLeastOneOf: []string{
+						"site_config.0.application_stack.0.dotnet_version",
+						"site_config.0.application_stack.0.php_version",
+						"site_config.0.application_stack.0.python_version",
+						"site_config.0.application_stack.0.node_version",
+						"site_config.0.application_stack.0.ruby_version",
+						"site_config.0.application_stack.0.java_version",
+						"site_config.0.application_stack.0.docker_image",
+					},
 					ValidateFunc: validation.StringInSlice([]string{
 						"12-lts",
 						"14-lts",
@@ -1145,6 +1229,15 @@ func linuxApplicationStackSchema() *pluginsdk.Schema {
 				"ruby_version": {
 					Type:     pluginsdk.TypeString,
 					Optional: true,
+					AtLeastOneOf: []string{
+						"site_config.0.application_stack.0.dotnet_version",
+						"site_config.0.application_stack.0.php_version",
+						"site_config.0.application_stack.0.python_version",
+						"site_config.0.application_stack.0.node_version",
+						"site_config.0.application_stack.0.ruby_version",
+						"site_config.0.application_stack.0.java_version",
+						"site_config.0.application_stack.0.docker_image",
+					},
 					ValidateFunc: validation.StringInSlice([]string{
 						"2.6",
 						"2.7",
@@ -1159,8 +1252,17 @@ func linuxApplicationStackSchema() *pluginsdk.Schema {
 				},
 
 				"java_version": {
-					Type:         pluginsdk.TypeString,
-					Optional:     true,
+					Type:     pluginsdk.TypeString,
+					Optional: true,
+					AtLeastOneOf: []string{
+						"site_config.0.application_stack.0.dotnet_version",
+						"site_config.0.application_stack.0.php_version",
+						"site_config.0.application_stack.0.python_version",
+						"site_config.0.application_stack.0.node_version",
+						"site_config.0.application_stack.0.ruby_version",
+						"site_config.0.application_stack.0.java_version",
+						"site_config.0.application_stack.0.docker_image",
+					},
 					ValidateFunc: validation.StringIsNotEmpty, // There a significant number of variables here, and the versions are not uniformly formatted.
 					// TODO - Needs notes in the docs for this to help users navigate the inconsistencies in the service. e.g. jre8 va java8 etc
 					ConflictsWith: []string{
@@ -1188,8 +1290,17 @@ func linuxApplicationStackSchema() *pluginsdk.Schema {
 				},
 
 				"docker_image": {
-					Type:         pluginsdk.TypeString,
-					Optional:     true,
+					Type:     pluginsdk.TypeString,
+					Optional: true,
+					AtLeastOneOf: []string{
+						"site_config.0.application_stack.0.dotnet_version",
+						"site_config.0.application_stack.0.php_version",
+						"site_config.0.application_stack.0.python_version",
+						"site_config.0.application_stack.0.node_version",
+						"site_config.0.application_stack.0.ruby_version",
+						"site_config.0.application_stack.0.java_version",
+						"site_config.0.application_stack.0.docker_image",
+					},
 					ValidateFunc: validation.StringIsNotEmpty,
 					RequiredWith: []string{
 						"site_config.0.application_stack.0.docker_image_tag",
@@ -2778,7 +2889,7 @@ func ExpandSiteConfigWindows(siteConfig []SiteConfigWindows, existing *web.SiteC
 		expanded.AppCommandLine = utils.String(winSiteConfig.AppCommandLine)
 	}
 
-	if metadata.ResourceData.HasChange("site_config.0.application_stack") {
+	if metadata.ResourceData.HasChange("site_config.0.application_stack") && len(winSiteConfig.ApplicationStack) > 0 {
 		winAppStack := winSiteConfig.ApplicationStack[0]
 		expanded.NetFrameworkVersion = utils.String(winAppStack.NetFrameworkVersion)
 		expanded.PhpVersion = utils.String(winAppStack.PhpVersion)

